wolfssh/log.h

```diff
@@ -47,20 +47,24 @@ enum wolfSSH_LogLevel {
 
 typedef void (*wolfSSH_LoggingCb)(enum wolfSSH_LogLevel,
                                   const char *const logMsg);
+WOLFSSH_API void wolfSSH_SetLoggingCb(wolfSSH_LoggingCb logF);
+WOLFSSH_API int wolfSSH_LogEnabled(void);
 
-WOLFSSH_API int  wolfSSH_SetLoggingCb(wolfSSH_LoggingCb logF);
 
-
-#ifdef DEBUG_WOLFSSH
-    WOLFSSH_API void WLOG(enum wolfSSH_LogLevel,const char *const logMsg, ...)
-    #ifdef __GNUC__
-        __attribute__((format(printf, 2, 3)));
-    #else
-        ;  /* end decl */
-    #endif /* __GNUC__ */
+#ifdef __GNUC__
+    #define FMTCHECK __attribute__((format(printf,2,3)))
 #else
-    #define WLOG(a, b, ...)
-#endif /* DEBUG_WOLFSSH */
+    #define FMTCHECK
+#endif /* __GNUC__ */
+
+
+WOLFSSH_API void wolfSSH_Log(enum wolfSSH_LogLevel,
+                             const char *const, ...) FMTCHECK;
+
+#define WLOG(...) do { \
```

Comment:
Previously WLOG could be compiled out. Why the transition to always calling the function wolfSSH_LogEnabled in the case where DEBUG_WOLFSSH is not defined?

Reply:
It'll get optimized out in this case. This ensures any changes won't break the logging. The code is still checked by the compiler, despite the inline function indicating that logging is disabled always returning that. SSH likes to have some things logged whether or not our debug logging is enabled.